# Outbreak Readiness System

In [2]:
import numpy as np
import pandas as pd
import json
#from dbfread import DBF
import geopandas as gpd
from explore import explore

## 2. Data Understanding

### 2.1. Kenya_hxl.csv dataset

#### 2.1.1. Loading the file

In [8]:
hxl_df = pd.read_csv("./Datasets/kenya_hxl.csv")
hxl_df.head(10)

,X,Y,osm_id,osm_type,completeness,#loc+amenity,#meta+healthcare,#loc +name,#meta +operator,#geo+bounds+url,...,#meta+is_in_health_zone,#contact +url,addr_housenumber,addr_street,addr_postcode,addr_city,changeset_id,changeset_version,changeset_timestamp,#meta +id
0,NaN,NaN,474439689,way,21.875,hospital,counselling,Community Health Center,MOH,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,170715105,5,2025/08/20 09:49:32+00,1e84916be0a9432a8b98d539bc014957
1,NaN,NaN,1224725709,way,15.625,hospital,hospital,JOHATTLAS MEDICAL AND DENTAL CENTER,NaN,microsoft/BuildingFootprints,...,NaN,NaN,NaN,NaN,NaN,NaN,163982188,2,2025/03/23 11:25:46+00,020c4dec61384d65853adb9ee4680784
2,36.887300,-1.254270,8891997083,node,12.500,*,alternative,Faith Healer,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,107380437,1,2021/07/04 08:44:18+00,4ac9577b8e094463b9436f95117a07c1
3,36.887851,-1.246880,2405823297,node,9.375,clinic,NaN,Redeemed Gospel Church Clinic,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,17201651,1,2013/08/03 09:20:18+00,71c100d49785492283c8c7475c7d172d
4,36.886524,-1.244329,8415761717,node,28.125,clinic,clinic,Access Afya Baba dogo,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,Nairobi,99124843,1,2021/02/11 17:31:32+00,3ad1a6e7f46649419f72c447d57ad7c7
5,38.273053,-1.384087,6244568075,node,9.375,hospital,NaN,Ngelani Dispensary,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,66740351,1,2019/01/29 14:30:47+00,a5fc107df3564044bcd9e39ca4180181
6,36.864206,-1.312513,12238169033,node,21.875,pharmacy,NaN,kwa grace chemist,NaN,survey,...,NaN,NaN,NaN,NaN,NaN,NaN,159324175,2,2024/11/19 13:13:51+00,293d5095a6db4ce9bac452eecdc188f5
7,NaN,NaN,792927093,way,21.875,hospital,counselling,mbauini_dispensary,government,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,170345419,2,2025/08/12 14:31:17+00,7d3d558e05b94e30a390f8d9ddf4d14e
8,NaN,NaN,667423495,way,25.000,hospital,counselling,Isiolo County Referal Hospital,public,NaN,...,NaN,NaN,NaN,Isiolo-Mandera Road,NaN,Isiolo,170430488,3,2025/08/14 10:57:47+00,b766a2d2cc6d4e0ead0bf94a49c7ddcf
9,NaN,NaN,232010702,way,9.375,hospital,NaN,Lions SightFirst Eye Hospital,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,159995187,3,2024/12/06 13:45:55+00,36b71ee7a7af435095edbc28f3b40ff5


#### 2.1.2. Checking the shape of the dataset

In [4]:
hxl_df.shape

(2640, 35)

**kenya_hxl** has **2,640** rows and **35** columns

#### 2.1.3. Checking for missing data

In [5]:
hxl_df.isnull().sum()

X                              721
Y                              721
osm_id                           0
osm_type                         0
completeness                     0
#loc+amenity                    32
#meta+healthcare              1231
#loc +name                     118
#meta +operator               2051
#geo+bounds+url               2464
#meta +speciality             2461
#meta +operator_type          1430
#contact +phone               2543
#status+operational_status    1922
#access +hours                2045
#capacity +beds               2510
#capacity +staff              2640
#meta +health_amenity_type    2622
#meta+dispensing              2256
#meta+wheelchair              2583
#meta+emergency               2506
#meta+insurance               2632
#meta+water_source            2623
#meta+electricity             2549
#meta+is_in_health_area       2640
#meta+is_in_health_zone       2640
#contact +url                 2637
addr_housenumber              2583
addr_street         

#### 2.1.4. Checking for datatypes

In [7]:
hxl_df.dtypes

X                             float64
Y                             float64
osm_id                          int64
osm_type                          str
completeness                  float64
#loc+amenity                      str
#meta+healthcare                  str
#loc +name                        str
#meta +operator                   str
#geo+bounds+url                   str
#meta +speciality                 str
#meta +operator_type              str
#contact +phone                   str
#status+operational_status        str
#access +hours                    str
#capacity +beds               float64
#capacity +staff              float64
#meta +health_amenity_type        str
#meta+dispensing                  str
#meta+wheelchair                  str
#meta+emergency                   str
#meta+insurance                   str
#meta+water_source                str
#meta+electricity                 str
#meta+is_in_health_area       float64
#meta+is_in_health_zone       float64
#contact +ur

#### 2.1.5. Checking the feature/column names

In [9]:
hxl_df.columns

Index(['X', 'Y', 'osm_id', 'osm_type', 'completeness', '#loc+amenity',
       '#meta+healthcare', '#loc +name', '#meta +operator', '#geo+bounds+url',
       '#meta +speciality', '#meta +operator_type', '#contact +phone',
       '#status+operational_status', '#access +hours', '#capacity +beds',
       '#capacity +staff', '#meta +health_amenity_type', '#meta+dispensing',
       '#meta+wheelchair', '#meta+emergency', '#meta+insurance',
       '#meta+water_source', '#meta+electricity', '#meta+is_in_health_area',
       '#meta+is_in_health_zone', '#contact +url', 'addr_housenumber',
       'addr_street', 'addr_postcode', 'addr_city', 'changeset_id',
       'changeset_version', 'changeset_timestamp', '#meta +id'],
      dtype='str')

### 2.2. Kenya.csv dataset

#### 2.2.1. Loading the file

In [10]:
kenya_df = pd.read_csv("./Datasets/kenya.csv")
kenya_df.head(10)

,X,Y,osm_id,osm_type,completeness,amenity,healthcare,name,operator,source,...,is_in_health_zone,url,addr_housenumber,addr_street,addr_postcode,addr_city,changeset_id,changeset_version,changeset_timestamp,uuid
0,NaN,NaN,474439689,way,21.875,hospital,counselling,Community Health Center,MOH,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,170715105,5,2025/08/20 09:49:32+00,1e84916be0a9432a8b98d539bc014957
1,NaN,NaN,1224725709,way,15.625,hospital,hospital,JOHATTLAS MEDICAL AND DENTAL CENTER,NaN,microsoft/BuildingFootprints,...,NaN,NaN,NaN,NaN,NaN,NaN,163982188,2,2025/03/23 11:25:46+00,020c4dec61384d65853adb9ee4680784
2,36.887300,-1.254270,8891997083,node,12.500,*,alternative,Faith Healer,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,107380437,1,2021/07/04 08:44:18+00,4ac9577b8e094463b9436f95117a07c1
3,36.887851,-1.246880,2405823297,node,9.375,clinic,NaN,Redeemed Gospel Church Clinic,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,17201651,1,2013/08/03 09:20:18+00,71c100d49785492283c8c7475c7d172d
4,36.886524,-1.244329,8415761717,node,28.125,clinic,clinic,Access Afya Baba dogo,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,Nairobi,99124843,1,2021/02/11 17:31:32+00,3ad1a6e7f46649419f72c447d57ad7c7
5,38.273053,-1.384087,6244568075,node,9.375,hospital,NaN,Ngelani Dispensary,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,66740351,1,2019/01/29 14:30:47+00,a5fc107df3564044bcd9e39ca4180181
6,36.864206,-1.312513,12238169033,node,21.875,pharmacy,NaN,kwa grace chemist,NaN,survey,...,NaN,NaN,NaN,NaN,NaN,NaN,159324175,2,2024/11/19 13:13:51+00,293d5095a6db4ce9bac452eecdc188f5
7,NaN,NaN,792927093,way,21.875,hospital,counselling,mbauini_dispensary,government,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,170345419,2,2025/08/12 14:31:17+00,7d3d558e05b94e30a390f8d9ddf4d14e
8,NaN,NaN,667423495,way,25.000,hospital,counselling,Isiolo County Referal Hospital,public,NaN,...,NaN,NaN,NaN,Isiolo-Mandera Road,NaN,Isiolo,170430488,3,2025/08/14 10:57:47+00,b766a2d2cc6d4e0ead0bf94a49c7ddcf
9,NaN,NaN,232010702,way,9.375,hospital,NaN,Lions SightFirst Eye Hospital,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,159995187,3,2024/12/06 13:45:55+00,36b71ee7a7af435095edbc28f3b40ff5


### 2.3. Kenya_hxl.geojson dataset

#### 2.3.1. Loading the file

In [ ]:
with open("./Datasets/kenya_hxl.geojson", "r") as file:
    hxl_json = json.load(file)

### 2.4. Kenya.geojson

#### 2.4.1. Loading the file

In [ ]:
with open("./Datasets/kenya.geojson", "r") as file:
    kenya_json = json.load(file)

### 2.5. Kenya.dbf

#### 2.5.1. Loading the file

In [11]:
dbf_df = gpd.read_file("/workspaces/Outbreak-Readiness-System/Datasets/kenya-shapefiles/shapefiles/kenya.dbf")
dbf_df.head(10)

,osm_id,osm_type,completene,amenity,healthcare,name,operator,source,speciality,operator_t,...,url,addr_house,addr_stree,addr_postc,addr_city,changeset_,changese_1,changese_2,uuid,geometry
0,8891997083,node,12.500,*,alternative,Faith Healer,NaN,NaN,NaN,NaN,...,None,NaN,NaN,NaN,NaN,107380437.0,1.0,2021-07-04,4ac9577b8e094463b9436f95117a07c1,POINT (36.8873 -1.25427)
1,2405823297,node,9.375,clinic,NaN,Redeemed Gospel Church Clinic,NaN,NaN,NaN,NaN,...,None,NaN,NaN,NaN,NaN,17201651.0,1.0,2013-08-03,71c100d49785492283c8c7475c7d172d,POINT (36.88785 -1.24688)
2,8415761717,node,28.125,clinic,clinic,Access Afya Baba dogo,NaN,NaN,NaN,private,...,None,NaN,NaN,NaN,Nairobi,99124843.0,1.0,2021-02-11,3ad1a6e7f46649419f72c447d57ad7c7,POINT (36.88652 -1.24433)
3,6244568075,node,9.375,hospital,NaN,Ngelani Dispensary,NaN,NaN,NaN,NaN,...,None,NaN,NaN,NaN,NaN,66740351.0,1.0,2019-01-29,a5fc107df3564044bcd9e39ca4180181,POINT (38.27305 -1.38409)
4,12238169033,node,21.875,pharmacy,NaN,kwa grace chemist,NaN,survey,NaN,private,...,None,NaN,NaN,NaN,NaN,159324175.0,2.0,2024-11-19,293d5095a6db4ce9bac452eecdc188f5,POINT (36.86421 -1.31251)
5,13965400039,node,12.500,hospital,hospital,EGESA MEDICAL CLINIC,NaN,NaN,NaN,NaN,...,None,NaN,NaN,NaN,NaN,184592113.0,1.0,2026-06-25,7850cb35c25c49ea80249b42065d9b96,POINT (34.78212 -0.65983)
6,1446059176,node,12.500,pharmacy,NaN,Faith medical clinic,NaN,NaN,NaN,NaN,...,None,NaN,NaN,NaN,NaN,18671139.0,2.0,2013-11-02,2d784a271f7f420a9a8093e8e72e18e4,POINT (36.86638 -1.3166)
7,8074036802,node,15.625,pharmacy,NaN,Galex,NaN,NaN,NaN,unknown,...,None,NaN,NaN,NaN,NaN,93408841.0,1.0,2020-11-02,d911b6c3d51b41cf8084333ea267d486,POINT (36.74841 -1.29171)
8,8074036784,node,15.625,pharmacy,NaN,Duka LA Dawa,NaN,NaN,NaN,unknown,...,None,NaN,NaN,NaN,NaN,93408841.0,1.0,2020-11-02,d32772adf4d041909181dae408876884,POINT (36.73329 -1.29145)
9,9487237401,node,12.500,hospital,hospital,Mbeere subcounty Hospital,NaN,NaN,NaN,NaN,...,None,NaN,NaN,NaN,NaN,117126652.0,1.0,2022-02-07,8f52d4f0b530469ba8a357d9f4054312,POINT (37.64251 -0.57672)
